src/main/java/com/squareup/kotlinpoet/Util.kt

```diff
@@ -23,14 +23,17 @@ internal object NullAppendable : Appendable {
   override fun append(c: Char) = this
 }
 
-internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V>
-    = Collections.unmodifiableMap(LinkedHashMap(this))
+internal fun <K, V> Map<K, V>.toImmutableMap(): Map<K, V> =
+    Collections.unmodifiableMap(LinkedHashMap(this))
 
-internal fun <T> Collection<T>.toImmutableList(): List<T>
-    = Collections.unmodifiableList(ArrayList(this))
+internal fun <T> Collection<T>.toImmutableList(): List<T> =
+    Collections.unmodifiableList(ArrayList(this))
 
-internal fun <T> Collection<T>.toImmutableSet(): Set<T>
-    = Collections.unmodifiableSet(LinkedHashSet(this))
+internal fun <T> Collection<T>.toImmutableSet(): Set<T> =
+    Collections.unmodifiableSet(LinkedHashSet(this))
+
+internal inline fun <reified T : Enum<T>> Collection<T>.toEnumSet(): Set<T> =
+    enumValues<T>().filter(this::contains).toSet()
```

Comment:
could be `enumValues<T>().filterTo(mutableSetOf(), this::contains)` to avoid the extra ArrayList.

Reply:
Nice, I'll use it!